packages/api-server/src/db/helpers.ts

```diff
@@ -223,75 +221,21 @@ export function filterLogsByAddress(
   return result;
 }
 
-export enum QueryRoundStatus {
-  keepGoing,
-  stop,
-}
-
-export interface ExecuteOneQueryResult {
-  status: QueryRoundStatus;
-  data: any[];
-}
-
 export function getDatabaseRateLimitingConfiguration() {
   const MAX_QUERY_NUMBER = envConfig["maxQueryNumber"]
     ? +envConfig["maxQueryNumber"]
     : DEFAULT_MAX_QUERY_NUMBER;
+  // TODO set query timeout
```

Comment:
can we add query time limit in this pr as well? this is important after db grows. I think simple solution like `Promise.race([query, timeout])` should be enough?

Reply:
Knex has a `timeout` method:

http://knexjs.org/guide/query-builder.html#timeout

If it actually cancels the query then it's better than `Promise.race`.

Reply:
> http://knexjs.org/guide/query-builder.html#timeout

cool this is more clean

Reply:
Thanks, @sopium @RetricSu 

The new commit f7771bbd653765382aff6dd3cc0fa833258b32d8 uses http://knexjs.org/guide/query-builder.html#timeout to add query  timeout limit. PTAL.